resolve.fallback fails for linked packages

<!-- Please don't delete this template because we'll close your issue -->
<!-- Before creating an issue please make sure you are using the latest version of webpack. -->

# Bug report

<!-- Please ask questions on StackOverflow or the webpack Gitter. -->
<!-- https://stackoverflow.com/questions/ask?tags=webpack -->
<!-- https://gitter.im/webpack/webpack -->
<!-- Issues which contain questions or support requests will be closed. -->

**What is the current behavior?**

Configuring `resolve.fallaback` does not work for modules linked from outside the package directory that webpack is being used from.  For example with the following file structure

```
root/
  +- pkg-a/
  |  +- package.json
  |  +- index.js (containing require("path"))
  +- pkg-b/
     +- package.json (containing dependency "pkg-a": "file:../pkg-a", webpack, webplack-cli and path-browserify)
     +- src/
     |  +- index.js (containg require("pkg-a"))
     +- webpack.config.js (containing resolve.fallback.path: "path-browserify")
```
Webpack fails to build with the error
```
assets by status 252 bytes [cached] 1 asset
./src/index.js 59 bytes [built] [code generated]
../pkg-a/index.js 30 bytes [built] [code generated]

ERROR in ../pkg-a/index.js 1:13-28
Module not found: Error: Can't resolve 'path' in 'webpack-bug/pkg-a'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add an fallback 'resolve.fallback: { "path": "path-browserify" }'
        - install 'path-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "path": false }
 @ ./src/index.js 2:10-26

webpack 5.0.0-rc.4 compiled with 1 error in 203 ms
```

This file structure is similar to how monorepos with tools like lerna are set up.

**If the current behavior is a bug, please provide the steps to reproduce.**

<!-- A great way to do this is to provide your configuration via a GitHub repository -->
<!-- The most helpful is a minimal reproduction with instructions on how to reproduce -->
<!-- Repositories with too many files or large `webpack.config.js` files are not suitable -->
<!-- Please only add small code snippets directly into this issue -->
<!-- https://gist.github.com is a good place for longer code snippets -->
<!-- If your issue is caused by a plugin or loader, please create an issue on the loader/plugin repository instead -->

See [this repository](https://github.com/Hornwitser/webpack-bug) for a minimal reproduction example.  Note that if pkg-a is moved into pkg-b/node_modules the build succeed.

**What is the expected behavior?**

<!-- "It should work" is not a helpful explanation -->
<!-- Explain exactly how it should behave -->

Module fallback should work irrespective of whether the package is linked from outside or installed into node_modules.

**Other relevant information:**
webpack version:  5.0.0-rc.4
Node.js version: v14.9.0
Operating System: Arch Linux
Additional tools: None.

--- a/lib/ModuleNotFoundError.js
+++ b/lib/ModuleNotFoundError.js
@@ -11,18 +11,18 @@ const WebpackError = require("./WebpackError");
 /** @typedef {import("./Module")} Module */
 
 const previouslyPolyfilledBuiltinModules = {
-	assert: "assert",
-	buffer: "buffer",
+	assert: "assert/",
+	buffer: "buffer/",
 	console: "console-browserify",
 	constants: "constants-browserify",
 	crypto: "crypto-browserify",
 	domain: "domain-browser",
-	events: "events",
+	events: "events/",
 	http: "stream-http",
 	https: "https-browserify",
 	os: "os-browserify/browser",
 	path: "path-browserify",
-	punycode: "punycode",
+	punycode: "punycode/",
 	process: "process/browser",
 	querystring: "querystring-es3",
 	stream: "stream-browserify",
@@ -31,12 +31,12 @@ const previouslyPolyfilledBuiltinModules = {
 	_stream_readable: "readable-stream/readable",
 	_stream_transform: "readable-stream/transform",
 	_stream_writable: "readable-stream/writable",
-	string_decoder: "string_decoder",
-	sys: "util",
+	string_decoder: "string_decoder/",
+	sys: "util/",
 	timers: "timers-browserify",
 	tty: "tty-browserify",
-	url: "url",
-	util: "util",
+	url: "url/",
+	util: "util/",
 	vm: "vm-browserify",
 	zlib: "browserify-zlib"
 };
@@ -63,14 +63,10 @@ class ModuleNotFoundError extends WebpackError {
 					"BREAKING CHANGE: " +
 					"webpack < 5 used to include polyfills for node.js core modules by default.\n" +
 					"This is no longer the case. Verify if you need this module and configure a polyfill for it.\n\n";
-				if (request !== alias) {
-					message +=
-						"If you want to include a polyfill, you need to:\n" +
-						`\t- add an fallback 'resolve.fallback: { "${request}": "${alias}" }'\n` +
-						`\t- install '${dependency}'\n`;
-				} else {
-					message += `If you want to include a polyfill, you need to install '${dependency}'.\n`;
-				}
+				message +=
+					"If you want to include a polyfill, you need to:\n" +
+					`\t- add a fallback 'resolve.fallback: { "${request}": require.resolve("${alias}") }'\n` +
+					`\t- install '${dependency}'\n`;
 				message +=
 					"If you don't want to include a polyfill, you can use an empty module like this:\n" +
 					`\tresolve.fallback: { "${request}": false }`;

